portal/src/main/webapp/js/src/dashboard/iviz.js

```diff
@@ -8178,3 +8179,199 @@ window.LogRankTest = (function(jStat) {
+                                                            saveCohort = true;
+                                                        }
+                                                    } else if (previousCase.samples.length === 
+                                                        selectedCase.samples.length) {
+                                                        var differentSamples = _.difference(previousCase.samples,
+                                                            selectedCase.samples);
+                                                        if (differentSamples.length > 0) {
+                                                            saveCohort = true;
+                                                        }
+                                                    }
+                                                }
+                                                return !saveCohort;
+                                            });
+                                        }
+                                        
+
+                                        if (saveCohort) {
+                                            var _selectedSamplesNum = 0;
+                                            var _selectedPatientsNum = 0;
+                                            _.each(self_.stats.selectedCases, function (studyCasesMap) {
+                                                _selectedSamplesNum += studyCasesMap.samples.length;
+                                                _selectedPatientsNum += studyCasesMap.patients.length;
+                                            });
+                                            self_.selectedSamplesNum = _selectedSamplesNum;
+                                            self_.selectedPatientsNum = _selectedPatientsNum;
+
+                                            vcSession.events.saveCohort(self_.stats,
+                                                cohortName, cohortDescription || '')
+                                                .done(function (response) {
+                                                    self_.savedVC = response;
+                                                    tooltip.find('.cohort-link').html(
+                                                    '<a class="virtual-study-link" href="' + window.cbioURL + 
+                                                        'study?id=' + self_.savedVC.id + '" onclick="window.open(\'' + 
+                                                        window.cbioURL + 'study?id=' + self_.savedVC.id + '\')">' + 
+                                                        window.cbioURL + 'study?id=' + self_.savedVC.id + '</a>');
+                                                    _.each(self_.stats.selectedCases, function(study){
+                                                       previousSelectedCases[study.studyID] = study; 
```

Comment:
you've confirmed this works? if `selectedCases` changes in place, then the `study` object will not hold onto the previous state, so it will never create a new URL. The way to fix this would be to do a deep copy of `study`